src/http/request.cr

```diff
@@ -11,7 +11,8 @@ class HTTP::Request
   @query_params : Params?
   @uri : URI?
 
-  def initialize(@method : String, @resource : String, @headers : Headers = Headers.new, @body = nil, @version = "HTTP/1.1")
+  def initialize(@method : String, @resource : String, headers : Headers? = nil, @body = nil, @version = "HTTP/1.1")
+    @headers = Headers.new_or_dup(headers)
```

Comment:
Can we maybe just do `headers.try(&.dup) || Headers.new`? I'm not sure we need an extra method just for this.